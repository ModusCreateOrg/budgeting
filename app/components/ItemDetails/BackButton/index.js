import React from 'react';

const BackButton = (props) => {
    return (
        <div>
            <button onClick={props.history.goBack}>Back</button>
        </div>
    );
}

export default BackButton;