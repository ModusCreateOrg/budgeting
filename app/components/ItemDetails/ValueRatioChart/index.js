import React from 'react';
import DonutChart from 'components/DonutChart';
import {ItemDetails} from 'modules/items';

const ValueRatioChart = ({transaction, totalFlowValue} : ItemDetails) => {
    
    const data = [{...transaction, value : Math.abs(transaction.value)}, {description : "Other items", value : Math.abs(totalFlowValue - transaction.value)}];

    return (
        <DonutChart data={data} dataLabel="description" dataKey="description"/>
    );
}

export default ValueRatioChart;