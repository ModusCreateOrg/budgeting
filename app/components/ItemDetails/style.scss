@import 'theme/variables';

.title {
  margin: 0;
  color: $black;
}
.neg {
  color: $red;
}
.pos {
  color: $green;
}
.backBtn {
  align-self: flex-end;
  padding: 5px;
  border-radius: 6px;
  font-size: 16px;
  color: $gray;
  border: 1px solid $lightgray;
  text-decoration: none;
}
