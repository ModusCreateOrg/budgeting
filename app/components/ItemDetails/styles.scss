@import 'theme/variables';

.backButton {
  padding: 10px;
  border: 1px solid $lightgray;
  border-radius: 6px;
  font-size: 16px;
  font-weight: 600;
}

.title {
  text-align: center;
  h2 {
    text-decoration: underline;
    font-size: 30px;
    letter-spacing: 2px;
  }
}

.positive {
  color: $green;
}

.negative {
  color: $red;
}
