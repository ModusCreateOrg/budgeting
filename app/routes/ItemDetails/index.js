import React from 'react';

import Chunk from 'components/Chunk';

const loadItemDetails = () => import('containers/ItemDetails');

const ItemDetails = () => <Chunk load={loadItemDetails} />;

export default ItemDetails;
