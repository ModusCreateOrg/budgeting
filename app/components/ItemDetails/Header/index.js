import React from 'react';
import styles from './styles.scss';
import { ItemDetails } from 'modules/items';

const ItemDetailsHeader = ({ transaction: { description }, ratio, isInflow }: ItemDetails) => {
    const sign = isInflow ? "+" : "-";
    
    return (
        <div>
            <h1>Item details for {description}</h1>
            <h3><span className={isInflow ? styles.inflow : styles.outflow}>{sign}</span>{ratio}%</h3>
        </div>
    );
}

export default ItemDetailsHeader;