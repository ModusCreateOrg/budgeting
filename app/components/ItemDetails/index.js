// @flow
import React from 'react';

import DonutChart from 'components/DonutChart';

import styles from './style.scss';

const ItemDetails = ({ transaction, amount, balance, history }) => {
  const DonutData = [...transaction, ...balance];

  return (
    <section>
      <div
        role="link"
        tabIndex="0"
        onClick={() => {
          history.goBack();
        }}
        className={styles.backBtn}
      >
        Back
      </div>
      <h3 className={styles.title}>{transaction.description}</h3>
      <span className={amount.isNegative ? styles.neg : styles.pos}>{amount.percentage}%</span>
      <DonutChart data={DonutData} dataLabel="description" dataKey="id" />
    </section>
  );
};

export default ItemDetails;
