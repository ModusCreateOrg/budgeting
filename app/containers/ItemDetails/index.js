// @flow
import * as React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import type { TransactionContribution } from 'selectors/transactions';
import { getTransactionContribution } from 'selectors/transactions';

import DonutChart from 'components/DonutChart';

import styles from './style.scss';

type ItemDetailsProps = {
  data: TransactionContribution,
};

class ItemDetails extends React.Component<ItemDetailsProps> {
  render() {
    const { data } = this.props;

    // Make chart data to show this item and rest transaction amounts
    const donutData = [
      {
        id: data.id,
        value: data.value,
        description: data.description,
      },
      {
        id: data.id + 1,
        value: data.totalTransactionOfFlow - data.value,
        description: 'Others',
      },
    ];

    return (
      <div className={styles.itemDetails}>
        <Link to="/budget">Go Back</Link>
        <h2>{data.description}</h2>
        <h3>
          {data.value < 0 ? <span className={styles.red}>-</span> : <span className={styles.green}>+</span>}
          {data.percentage}%
        </h3>
        <DonutChart data={donutData} dataLabel="description" dataKey="id" />
      </div>
    );
  }
}

const mapStateToProps = (state, props) => ({
  data: getTransactionContribution(state, parseInt(props.match.params.itemId, 10)),
});

export default connect(mapStateToProps)(ItemDetails);
