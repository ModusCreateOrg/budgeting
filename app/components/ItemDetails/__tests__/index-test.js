import React from 'react';
import renderer from 'react-test-renderer';
import ItemDetails from 'components/ItemDetails';

jest.mock('components/DonutChart', () => 'div');

it('renders correctly', () => {
  const mockTransaction = {
    id: 1,
    description: 'The usual weekly run',
    itemOperator: '-',
    percent: 23.98,
    value: 1100,
    categoryId: 1,
    totalBudget: 4588.07,
    flow: 'Outflow',
  };

  const tree = renderer.create(<ItemDetails transaction={mockTransaction} />).toJSON();
  expect(tree).toMatchSnapshot();
});
