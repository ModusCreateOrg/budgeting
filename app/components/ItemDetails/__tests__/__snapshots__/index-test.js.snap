// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section>
  <button
    className="backButton"
    onClick={[Function]}
  >
    Go Back
  </button>
  <div
    className="title"
  >
    <h2>
      The usual weekly run
    </h2>
    <h3>
      <span
        className="negative"
      >
        -
      </span>
      23.98
      %
    </h3>
  </div>
  <div
    className="detail"
  >
    <div
      data={
        Array [
          Object {
            "categoryId": "0",
            "key": 0,
            "label": "Outflow Budget",
            "value": 3488.0699999999997,
          },
          Object {
            "categoryId": "1",
            "key": 1,
            "label": "The usual weekly run",
            "value": 1100,
          },
        ]
      }
      dataKey="key"
      dataLabel="label"
    />
  </div>
</section>
`;
